test/Feature/InsertSpec.hs

```diff
@@ -135,18 +135,14 @@ spec actualPgVersion = do
 
     context "with no pk supplied" $ do
       context "into a table with auto-incrementing pk" $
-        it "succeeds with 201 and link" $ do
-          p <- post "/auto_incrementing_pk" [json| { "non_nullable_string":"not null"} |]
-          liftIO $ do
-            simpleBody p `shouldBe` ""
-            simpleHeaders p `shouldSatisfy` matchHeader hLocation "/auto_incrementing_pk\\?id=eq\\.[0-9]+"
-            simpleStatus p `shouldBe` created201
-          let Just location = lookup hLocation $ simpleHeaders p
-          r <- get location
```

Comment:
Sometimes rows were confirmed to exist or not exist anymore in the same tests with additional requests. Those failed with the rolled back main request.

Some of those, like in this case, could safely be removed. This test is about status code and location header. The location header is now hardcoded instead of a regex (because it's now predictable which ID it will get), so the location URL does not need to be confirmed anymore.

Reply:
This one is much more readable now. I prefer the hardcoded Location.